src/components/Container.tsx

```diff
@@ -0,0 +1,34 @@
+import { ComponentPropsWithoutRef, ReactNode } from "react";
+import styled, { CSSProp, CSSProperties } from "styled-components";
+
+interface Props extends ComponentPropsWithoutRef<"div"> {
+  children?: ReactNode;
+  minWidth?: CSSProperties["minWidth"];
+  maxWidth?: CSSProperties["maxWidth"];
+  etc?: CSSProp;
+}
```

Comment:
아래 내용은 PR에서 권장하는 리뷰는 아니라서 그냥 제 생각대로 한 가지 추천드리는 내용입니다!! 가볍게 읽어주세욥

<img width="644" alt="image" src="https://github.com/woowacourse/layout-component/assets/56749516/528595d2-bd8f-4792-aedc-f392f3b463f3">

index.d.ts을 근거로, 다음과 같이도 쓸 수 있을 것 같아요!

```suggestion

interface Props extends ComponentPropsWithoutRef<"div">, PropsWithChildren {
  minWidth?: CSSProperties["minWidth"];
  maxWidth?: CSSProperties["maxWidth"];
  etc?: CSSProp;
}
```

이어서, 아래 스타일 컴포넌트에서도 해당 타입이 사용되는 것 같은데, 저는 변경 사항이 생겼을 경우에 같은 속성을 가지는 타입의 중복이 있으므로 개발자가 두 번 고쳐주어야 하는 번거로움이 있고 또 모를 휴먼 에러가 발생할 수 있다고 생각해 다음과 같이 고쳐보는 것도 추천 드립니다!
```tsx
interface ContainerProps {
    minWidth?: CSSProperties["minWidth"];
    maxWidth?: CSSProperties["maxWidth"];
    etc?: CSSProp;
}

type Props = ContainerProperty & ComponentPropsWithoutRef<"div"> & PropsWithChildren

// ... 

const S = {
  Wrapper: styled.div<ContainerProps>`
    min-width: ${({ $minWidth }) => $minWidth};
    max-width: ${({ $maxWidth }) => $maxWidth};
    margin: 0 auto;
    ${({ $etc }) => $etc}
  `,
};
```

Reply:
children을 직접 추가하는 것 보다 `PropsWithChildren`을 확장하는게 더 깔끔한 것 같네요, 적용해야겠어요!
다만, 추천해 주신 방법으로 중복을 줄여보려 했는데,

아래처럼 대문자를 쓴다고 오류를 뱉는 관계로.. 좋은 해결법을 찾지 못했습니다. ㅠㅠ
<img width="550" alt="image" src="https://github.com/woowacourse/layout-component/assets/116625502/84ed0688-9dc8-43bb-864a-757879a7fb72">

Reply:
저와 겪은 같은 오류군요 ㅠㅠ 저는 lowercase로 해결하였습니다!!